src/ADAL.PCL/TokenCache.cs

```diff
@@ -359,13 +363,17 @@ internal AuthenticationResultEx LoadFromCache(CacheQueryData cacheQueryData, Cal
                     bool tokenExtendedLifeTimeExpired = (resultEx.Result.ExtendedExpiresOn <=
                                             DateTime.UtcNow);
 
+                    cacheEvent.TokenNearExpiry = tokenNearExpiry.ToString();
+                    cacheEvent.TokenExtendedLifeTimeExpired = tokenExtendedLifeTimeExpired.ToString();
+                    cacheEvent.TokenExpired = (resultEx.Result.ExpiresOn <= DateTime.UtcNow).ToString();
                     //check for cross-tenant authority
                     if (!cacheKey.Authority.Equals(cacheQueryData.Authority))
                     {
                         // this is a cross-tenant result. use RT only
                         resultEx.Result.AccessToken = null;
                         PlatformPlugin.Logger.Information(callState,
                             "Cross Tenant refresh token was found in the cache");
+                        cacheEvent.IsCrossTenantRt = true.ToString();
```

Comment:
done.List of the parameters on the telemetry doc.

---
In reply to: [76154896](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/494#discussion_r76154896) [](ancestors = 76154896)